subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -761,6 +806,10 @@ static int wifi_iface_stats(uint32_t mgmt_request, struct net_if *iface,
 		return -ENOTSUP;
 	}
 
+	if (!net_if_is_admin_up(iface)) {
+		return -ENETDOWN;
+	}
```

Comment:
Also statistics could be shown for down interfaces like how `net stats` command does it.